File: JAVA/level2/124_나라의_숫자.java
class Solution {
    public String solution(int n) {
        String answer = "";

        while(n != 0){
            if(n%3 == 0)
                answer = "4" + answer;
            if(n%3 == 1)
                answer = "1" + answer;
            if(n%3 == 2)
                answer = "2" + answer;

            if(n%3 == 0)
                n = n/3 -1;
            else
                n /= 3;
        }
        return answer;
    }
}

File: JavaScript/level2/124_나라의_숫자.js
function solution(n) {
    var answer = '';
    
    while(n > 0){
        if(n%3 === 1){
            answer = "1" + answer;
            n = (n - n%3)/3;
        }
        else if(n%3 === 2){
            answer = "2" + answer;
            n = (n - n%3)/3;
        }
        else if(n%3 === 0){
            answer = "4" + answer;
            n = (n - 3)/3;
        }
    }
    
    return answer;
}